webrender_traits/src/api.rs

```diff
@@ -248,6 +248,10 @@ impl RenderApi {
         self.api_sender.send(msg).unwrap();
     }
 
+    pub fn shut_down(&self) {
```

Comment:
I was thinking along the following line:
  - add an `Arc<()>` field to always accompany `MsgSender`
  - hide `MsgSender` from public interfaces to enforce ^
  - when any owner of `MsgSender` gets destroyed, it can check if that dummy `Arc` reference is the last one
  - so that it can send the `Shutdown` signal before completely going down

This is a bit crappy/hacky, but it would work.